lib/WasmReader/WasmSections.h

```diff
@@ -3,17 +3,16 @@
 // Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
 //-------------------------------------------------------------------------------------------------------
 
-//          (name                 , ID                   , SectionFlag)
-WASM_SECTION(Signatures           , "signatures"         , fSectNone  )
-WASM_SECTION(ImportTable          , "import_table"       , fSectIgnore)
-WASM_SECTION(FunctionSignatures   , "function_signatures", fSectNone  )
-WASM_SECTION(IndirectFunctionTable, "function_table"     , fSectNone  )
-WASM_SECTION(Memory               , "memory"             , fSectNone  )
-WASM_SECTION(ExportTable          , "export_table"       , fSectNone  )
-WASM_SECTION(FunctionBodies       , "function_bodies"    , fSectNone  )
-WASM_SECTION(StartFunction        , "start_function"     , fSectIgnore)
-WASM_SECTION(DataSegments         , "data_segments"      , fSectNone  )
-WASM_SECTION(Names                , "names"              , fSectIgnore)
-WASM_SECTION(End                  , "end"                , fSectNone  )
+//          (name                 , ID                   , SectionFlag, Precedent         )
+WASM_SECTION(Signatures           , "signatures"         , fSectNone  , Invalid           )
+WASM_SECTION(ImportTable          , "import_table"       , fSectIgnore, Invalid           )
+WASM_SECTION(FunctionSignatures   , "function_signatures", fSectNone  , Signatures        )
+WASM_SECTION(IndirectFunctionTable, "function_table"     , fSectNone  , FunctionSignatures)
+WASM_SECTION(Memory               , "memory"             , fSectNone  , Invalid           )
+WASM_SECTION(ExportTable          , "export_table"       , fSectNone  , FunctionSignatures)
+WASM_SECTION(StartFunction        , "start_function"     , fSectIgnore, FunctionSignatures)
+WASM_SECTION(FunctionBodies       , "function_bodies"    , fSectNone  , FunctionSignatures)
+WASM_SECTION(DataSegments         , "data_segments"      , fSectNone  , Memory            )
+WASM_SECTION(Names                , "names"              , fSectIgnore, Invalid           )
```

Comment:
Apparently so, the end section is gone and not needed anyway with the fixed order and section length in the header